spec/posts_index_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let(:user) do
+    User.create(name: 'Naan', photo: 'https://unsplash.com/photos/ABCDEF', posts_counter: 5)
+  end
+
+  let!(:post) do
+    Post.create(title: 'My First Post', text: 'This is my first post', author: user, comments_counter: 0,
+                likes_counter: 0)
+  end
+
+  let!(:comment) do
+    Comment.create(text: 'This is my first comment', author: user, post:)
+  end
+
+  it "displays the user's name on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_content(user.name)
+  end
+
+  it "displays the user's post count on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_content(user.posts_counter)
+  end
+
+  it 'displays a post title on user posts page' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.title)
+  end
+
+  it 'displays a post text on user posts page' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.text)
+  end
+
+  it 'displays the first comments on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(comment.text)
+  end
```

Comment:
- [ ] @team a great job adding a test to check if the page displays the  first comments on a post, however, we are supposed to check if the `recent_comments` method is displaying only 5 recent comments

Reply:
@danmainah the requirement explicitly asks to create a test for "I can see the first comments on a post". There is no requirement to check if the method is displaying 5 recent comments.

![image](https://github.com/naanmohammed/Fintech-Forum/assets/124832435/e135aab6-3573-49ef-8adc-8e26c9cd8121)
![image](https://github.com/naanmohammed/Fintech-Forum/assets/124832435/48d9aeed-6190-4908-af4a-8184577f087f)

Reply:
@naanmohammed I agree, technically it doesn't show the user needs to check if the method displays 5 comments